src/test/java/com/github/dockerjava/cmd/PushImageCmdIT.java

```diff
@@ -50,31 +49,27 @@ public void pushLatest() throws Exception {
         String imageId = dockerRule.getClient().commitCmd(container.getId()).withRepository(username + "/busybox").exec();
 
         // we have to block until image is pushed
-        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback()).awaitSuccess();
+        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback())
+                .awaitCompletion(30, TimeUnit.SECONDS);
 
         LOG.info("Removing image: {}", imageId);
         dockerRule.getClient().removeImageCmd(imageId).exec();
 
-        dockerRule.getClient().pullImageCmd(username + "/busybox").exec(new PullImageResultCallback()).awaitSuccess();
+        dockerRule.getClient().pullImageCmd(username + "/busybox").exec(new PullImageResultCallback())
+                .awaitCompletion(30, TimeUnit.SECONDS);
     }
 
     @Test
     public void pushNonExistentImage() throws Exception {
-
-        if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_26)) {
-         // no errors??
-        } else if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_24)) {
-            exception.expect(DockerClientException.class);
-        } else {
-            exception.expect(NotFoundException.class);
+        //swarms throws a different error here
+        try {
+            dockerRule.getClient().pushImageCmd(username + "/xxx")
+                    .exec(new PushImageResultCallback())
+                    .awaitCompletion(30, TimeUnit.SECONDS);
+
+            Assert.fail("An exception is expected.");
```

Comment:
Assert.fail() is not junit Expected Rule style

Reply:
Yes sure is there a different easy way to define two different expectations?

Reply:
fast lookup shows that `ExpectedException.expect(Class)` uses `org.hamcrest.CoreMatchers#instanceOf` and there is 
so i guess you can construct `org.junit.rules.ExpectedException#expect(org.hamcrest.Matcher<?>)` + `org.hamcrest.CoreMatchers#anyOf(java.lang.Iterable<org.hamcrest.Matcher<? super T>>)` + instanceOf()

But again, you are changing the existing behaviour of the test. Test was about success + different api versions, so you shouldn't combine and hide this errors.

Reply:
That will make only sense for me if I can always expect the same exception for the same error. But that is not the case. If I upgrade my docker version, for example, I will get no a different Exception. Thats not a really cool behavior. And since docker behaves like that, it should be more straightforward to throw an exception that I can always handle. For instance a common DockerException base class for all Exception which the docker client will throw.